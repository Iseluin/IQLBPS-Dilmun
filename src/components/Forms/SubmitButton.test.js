import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react'
import { SubmitButton } from '.'

test('SubmitButton renders correctly', () => {
    const tree = renderer.create(<SubmitButton />).toJSON();
    expect(tree).toMatchSnapshot();
});

test('should have a disabled', () => {
    render(<SubmitButton disabled />);
    const input = screen.getByRole('button')
    expect(input).toBeDisabled();
})

test('should have a className', () => {
    render(<SubmitButton className="test" />);
    const input = screen.getByRole('button')
    expect(input).toHaveClass('test');
})

test('should have a style', () => {
    render(<SubmitButton style={{ color: 'red' }} />);
    const input = screen.getByRole('button')
    expect(input).toHaveStyle('color: red');
})

test('should have a id', () => {
    render(<SubmitButton id="test" />);
    const input = screen.getByRole('button')
    expect(input).toHaveAttribute('id', 'test');
})

test('should have a type', () => {
    render(<SubmitButton type="number" />);
    const input = screen.getByRole('button')
    expect(input).toHaveAttribute('type', 'number');
})

test('should have a onClick', () => {
    const onClick = jest.fn();
    render(<SubmitButton onClick={onClick} />);
    const input = screen.getByRole('button')
    input.click();
    expect(onClick).toHaveBeenCalled();
})

test('should have a text', () => {
    render(<SubmitButton buttonText='test' />);
    const input = screen.getByRole('button')
    expect(input).toHaveTextContent('test');
})
