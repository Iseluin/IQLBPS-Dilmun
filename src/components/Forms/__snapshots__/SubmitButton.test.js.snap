// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmitButton renders correctly 1`] = `
<button
  className="shadow-lg justify-center py-2 px-4 border border-secondary rounded-md text-sm font-medium text-background bg-primary hover:bg-primary focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-secondary"
  type="submit"
>
  undefined
</button>
`;
